solvers/jaxopt_gd.py

```diff
@@ -0,0 +1,146 @@
+from benchopt import BaseSolver
+from benchopt.stopping_criterion import SufficientProgressCriterion
+
+from benchopt import safe_import_context
+
+with safe_import_context() as import_ctx:
+    from benchmark_utils import constants
+    from benchmark_utils.learning_rate_scheduler import update_lr
+    from benchmark_utils.learning_rate_scheduler import init_lr_scheduler
+
+    import jax
+    import jax.numpy as jnp
+    from functools import partial
+
+    import jaxopt
+
+
+class Solver(BaseSolver):
+    """Two loops solver."""
+    name = 'jaxopt'
+
+    requirements = ["pip:jaxopt"]
+
+    stopping_criterion = SufficientProgressCriterion(
+        patience=constants.PATIENCE, strategy='callback'
+    )
+
+    # any parameter defined here is accessible as a class attribute
+    parameters = {
+        'inner_solver': ['gd', 'lbfgs'],
+        'step_size_outer': [1.],
+        'eval_freq': [1],
+        'n_inner_steps': [100],
+    }
+
+    @staticmethod
+    def get_next(stop_val):
+        return stop_val + 1
+
+    def set_objective(self, f_train, f_val, n_inner_samples, n_outer_samples,
+                      inner_var0, outer_var0):
+        self.n_inner_samples = n_inner_samples
+        self.n_outer_samples = n_outer_samples
+
+        self.f_inner = jax.jit(
+            partial(f_train(framework='jax'),
+                    batch_size=n_inner_samples, start=0),
+        )
+        self.f_outer = jax.jit(
+            partial(f_val(framework='jax'),
+                    batch_size=n_outer_samples, start=0),
+        )
+
+        @partial(jax.jit, static_argnames=("f", "n_steps"))
+        def inner_solver_fun(outer_var, inner_var, f=None, n_steps=1, lr=.1):
+            if self.inner_solver == 'gd':
+                solver = jaxopt.GradientDescent(
+                    fun=f, maxiter=n_steps, implicit_diff=True,
```

Comment:
Can you document what are the implicit diff parameters for these two solvers?

Reply:
cf #24